### PR TITLE
[4.0] Update composer deps to latest versions

### PR DESCRIPTION
**Pull request by wilsonge**
Update composer deps to latest + greatest allowed

- Remove lazy listener as this has been ported to the framework package (https://github.com/joomla-framework/event/pull/19)
- Small reworks as part of implementing the session aware trait (https://github.com/joomla-framework/application/pull/88)

### CODE REVIEW

**Comment by SharkyKZ** [comment]
Getting this in CLI:

```
Fatal error: Uncaught TypeError: Argument 1 passed to Symfony\Component\Debug\ExceptionHandler::handle() must be an instance of Exception, instance of Error given in C:\xampp\htdocs\joomla-cms\libraries\vendor\symfony\debug\ExceptionHandler.php:118
Stack trace:
#0 [internal function]: Symfony\Component\Debug\ExceptionHandler->handle(Object(Error))
#1 {main}
  thrown in C:\xampp\htdocs\joomla-cms\libraries\vendor\symfony\debug\ExceptionHandler.php on line 118

```

**Comment by brianteeman** [comment]
@wilsonge can you fix the conflicts please

**Comment by wilsonge** [comment]
Done and did another run of composer update for good measure - but this still won't fix the console package issues

**Comment by wilsonge** [comment]
@mbabker well I don't like what I've done here in any way - but I've gone and done it. Thoughts?

File: libraries/src/Application/ConsoleApplication.php [hunk 1]
```diff
@@ -87,6 +95,9 @@ public function __construct(
 		?Container $container = null
 	)
 	{
+		// Set up a Input object for Controllers etc to use
+		$this->input = new \Joomla\Input\Cli;
```
Comment (mbabker) [inline libraries/src/Application/ConsoleApplication.php]: CMS namespace, CLI was removed from the FW package.

File: libraries/src/Application/ConsoleApplication.php [hunk 2]
```diff
@@ -35,6 +35,14 @@ class ConsoleApplication extends Application implements DispatcherAwareInterface
 {
 	use DispatcherAwareTrait, EventAware, IdentityAware, ContainerAwareTrait, ExtensionManagerTrait, ExtensionNamespaceMapper;
 
+	/**
+	 * The input.
+	 *
+	 * @var    \Joomla\Input\Input
+	 * @since  4.0.0
+	 */
+	public $input = null;
```
Comment (mbabker) [inline libraries/src/Application/ConsoleApplication.php]: I’d follow the FW web app here; make this protected, add a `getInput()` method to satisfy that interface, and magic getter with direct property deprecation notice.

Comment (wilsonge) [reply libraries/src/Application/ConsoleApplication.php]: Good point - had forgotten about that somehow - done

File: libraries/src/Application/ConsoleApplication.php (hunk 1 above)
Comment (wilsonge) [reply libraries/src/Application/ConsoleApplication.php]: Still using 1.3 of input :P https://github.com/joomla/joomla-cms/blob/4.0-dev/composer.json#L67 but done in anticipation of that

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on cbc45b87986f06262d362b95c9e9eaa5a79ab9e1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26820">issues.joomla.org/tracker/joomla-cms/26820</a>.</sub>

**Comment by infograf768** [comment]
Something broken here:

Attempted to load class "LazyServiceEventListener" from namespace "Joomla\Event".
Did you forget a "use" statement for another namespace?

`        $listener = new LazyServiceEventListener($container, 'session.event_listener.metadata_manager', 'onAfterSessionStart');`

**Comment by SharkyKZ** [comment]
@infograf768 Run `composer install`.

**Comment by infograf768** [comment]
I did but I get an error (MAMP macintosh)
Problem 1
    - Installation request for phpdocumentor/reflection-docblock 5.1.0 -> satisfiable by phpdocumentor/reflection-docblock[5.1.0].
    - phpdocumentor/reflection-docblock 5.1.0 requires ext-filter ^7.1 -> the requested PHP extension filter has the wrong version (0.11.0) installed.
  Problem 2
    - phpdocumentor/reflection-docblock 5.1.0 requires ext-filter ^7.1 -> the requested PHP extension filter has the wrong version (0.11.0) installed.
    - phpspec/prophecy v1.10.2 requires phpdocumentor/reflection-docblock ^2.0|^3.0.2|^4.0|^5.0 -> satisfiable by phpdocumentor/reflection-docblock[5.1.0].
    - Installation request for phpspec/prophecy v1.10.2 -> satisfiable by phpspec/prophecy[v1.10.2].

trying to figure how to solve this...

**Comment by mbabker** [comment]
Use the `--ignore-platform-reqs` flag to bypass it.

**Pull request merged**